### PR TITLE
Site Settings: Use feature checks rather than plan checks

### PR DESCRIPTION
**Pull request by obenland**
#### Changes proposed in this Pull Request

* Replaced uses of `siteHasScanProductPurchase` with feature checks.
* Removes `siteHasScanProductPurchase` selector.

#### Testing instructions

* Navigate to /site-settings/jetpack
* Make sure you see the connection box on Atomic sites and Jetpack sites with scan feature image
* The changes in Navigation are only visible to sites on Jetpack below v9.9.

Related to p4TIVU-a66-p2
Depends on https://github.com/Automattic/wp-calypso/pull/63735

### CODE REVIEW

**Review by gavande1 — APPROVED** [review]
This works as expected. 👍

File: client/components/jetpack/sidebar/menu-items/jetpack-cloud.jsx [hunk 1]
```diff
@@ -20,7 +21,7 @@ export default ( { path } ) => {
 	const translate = useTranslate();
 	const siteId = useSelector( getSelectedSiteId );
 	const siteSlug = useSelector( getSelectedSiteSlug );
-	const hasScanProduct = useSelector( ( state ) => siteHasScanProductPurchase( state, siteId ) );
+	const hasScan = useSelector( ( state ) => siteHasFeature( state, siteId, WPCOM_FEATURES_SCAN ) );
 	const hasActiveRewind = useSelector( ( state ) => isRewindActive( state, siteId ) );
```
Comment (obenland) [inline client/components/jetpack/sidebar/menu-items/jetpack-cloud.jsx]: @bluefuton Checking the rewind status removes access to the settings menu for multisite jetpack sites that have backup purchases. Should this be a feature check for `backups` (2c728-pb/#plain) instead?

Comment (bluefuton) [reply client/components/jetpack/sidebar/menu-items/jetpack-cloud.jsx]: Thanks @obenland, yes, that sounds sensible 👍 

I based my logic on `client/my-sites/site-settings/settings-security/main.jsx` that you're also updating in this PR.

Comment (bluefuton) [reply client/components/jetpack/sidebar/menu-items/jetpack-cloud.jsx]: cc @Automattic/yamato-backup

Comment (obenland) [reply client/components/jetpack/sidebar/menu-items/jetpack-cloud.jsx]: > I based my logic on client/my-sites/site-settings/settings-security/main.jsx that you're also updating in this PR.

Should I replace `ActiveRewind` with a feature check for `backups` there as well?

Comment (bluefuton) [reply client/components/jetpack/sidebar/menu-items/jetpack-cloud.jsx]: It'd be good to have the Backup team's input on this - can you help @seear?

Comment (seear) [reply client/components/jetpack/sidebar/menu-items/jetpack-cloud.jsx]: > Should I replace ActiveRewind with a feature check for backups there as well?

Yes, the intention here would have been to check for backup.

**Review by DustyReagan — APPROVED** [review]
Tests well! 👍

**Pull request merged**